src/script/sigcache.cpp

```diff
@@ -26,7 +26,7 @@ class CSignatureCache
     CSHA256 m_salted_hasher;
     typedef CuckooCache::cache<uint256, SignatureCacheHasher> map_type;
     map_type setValid;
-    boost::shared_mutex cs_sigcache;
+    std::shared_mutex cs_sigcache;
```

Comment:
note: I think this has little risk of difference with boost vs std since we never use a condition variable with this mutex.

offtopic note for anyone curious is that cs_sigcache is overkill since we generally know that there are no concurrent readers and writers ever, the read lock could be held at a higher layer for the entire span of block validation, it's just harder to expose the right API for that (@hebasto seems to be looking at this stuff so figured I would mention it somewhere).